x-pack/plugins/cloud_security_posture/server/plugin.ts

```diff
@@ -78,16 +82,25 @@ export class CspPlugin
       ...plugins.fleet,
     });
 
-    initializeCspTransformsIndices(core.elasticsearch.client.asInternalUser, this.logger);
-    plugins.fleet.fleetSetupCompleted().then(() => {
+    plugins.fleet.fleetSetupCompleted().then(async () => {
+      const packageInfo = await plugins.fleet.packageService.asInternalUser.getInstallation(
+        CIS_KUBERNETES_PACKAGE_NAME
+      );
+
+      // If package is installed we want to make sure all needed assets are installed
+      if (packageInfo) {
+        this.initialize(core);
+      }
+
       plugins.fleet.registerExternalCallback(
         'packagePolicyPostCreate',
         async (
           packagePolicy: PackagePolicy,
           context: RequestHandlerContext,
-          request: KibanaRequest
+          _: KibanaRequest
         ): Promise<PackagePolicy> => {
           if (packagePolicy.package?.name === CIS_KUBERNETES_PACKAGE_NAME) {
+            await this.initialize(core);
```

Comment:
nit:
I would consider consolidating all the post create functions into one method, so maybe `initialize` should include `onPackagePolicyPostCreateCallback` as well? 
wdyt?

Reply:
the callback of `onPackagePolicyPostCreateCallback` is not part of the `initialize` of the plugin.
how do you see they relate?